### PR TITLE
Nav Block: Show justification controls for vertical variant

### LINKED ISSUE #29459: Navigation: Add alignment option to vertical variant
Navigation block offers two variants - horizontal and vertical. The vertical variant is missing alignment options. Space-between will probably not apply, but it would be useful to introduce the left/center/right.

Here are the settings in horizontal variant:
image

### PR DESCRIPTION
**Pull request by mkaz**
## Description

- Enable justify content controls to show for both horiz and vert variants
- Filter out space-between from vertical controls
 
Closes #29459 

## How has this been tested?

1. Add Navigation block (vertical)
2. Confirm the left-center-right controls show and work as expected
3. Confirm the space-between controls do not show
4. Test Navigation block (horizontal) space-between shows and no regression

## Screenshots 

See comments below.

### CODE REVIEW

**Comment by jasmussen** [comment]
Thank you Marcus, it's already been a long day for me so I'm going to take a look at this one tomorrow!

**Comment by jasmussen** [comment]
Fixed the alignments. Let's get a sanity check and merge.

now

**Comment by mkaz** [comment]
@jasmussen Ah, right `flex-direction: column` for vertical. I knew I was missing something straight forward.

Testing it out and it doesn't seem to be working properly for horizontal, the vertical list looks and works fine but horizontal changes are not being seen with the block selected, but only show when not selected. See GIF below,

alignment-not-selected

**Comment by jasmussen** [comment]
Try deselecting the horizontal version once you apply an alignment. If that works, then it's a known issue partially mitigated by https://github.com/WordPress/gutenberg/pull/29771

Edit: ah I see you did try deselecting. Yeah that should be present also in trunk.

**Comment by mkaz** [comment]
@jasmussen Confirmed, same behavior exists on trunk! No regression on this PR :heavy_check_mark:

**Comment by mkaz** [comment]
I did more testing on other themes and in different regiions, both the Navigation Editor, and Site Editor Screens.

vert-nav-site-editor

vert-nav-editor-screen

**Comment by jasmussen** [comment]
Awesome testing. Actually the fact that it affects the navigation screen — it probably shouldn't, which is to say @talldan might want to know.

**Comment by mkaz** [comment]
@jasmussen Good catch on the navigation editor screen! I tested it there and confirmed it worked, but didn't think if it makes sense there or not. Considering that screen is to create and edit menus for classic themes, the layout format won't work and should not be there.

In 9ebbbab, I added a new attribute that can be used to disable the controls, and updated the Nav Editor to pass in this attribute so the justify controls will not show. Tested and confirmed all are showing everywhere else, but not there. 👍 

@talldan Can you confirm this makes sense, and works as you would expect? 🙏

**Comment by talldan** [comment]
@mkaz Thanks for the ping. @jasmussen is right in that it shouldn't be on the navigation screen.

This navigation screen previously disabled this using a prop that's passed into the block's edit function, but it looks like this PR removes that prop:
https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-navigation/src/filters/remove-edit-unsupported-features.js#L17

Instead of removing it, you should be able to default it to `true` so that it shows in both orientations, which the nav editor will then override.

I think it'd be good to avoid using an attribute for this purpose, as attributes shouldn't really be for editor side settings, more for the frontend representation of a block.

Hope that helps.

**Comment by talldan** [comment]
BTW - there is an issue to discuss better ways of enabling/disabling features in the nav block (https://github.com/WordPress/gutenberg/issues/30007).

I think content justification is one of the easier ones, in that it could be a candidate for becoming a block 'supports' feature. I know there's something very similar on Buttons block as well. One thing to check is that the changes here don't diverge too much from the Buttons implementation.

**Comment by mkaz** [comment]
@talldan After looking into it and rediscovering why we opted for this implementation in the first place. The way the justification works is it applies to the inner elements and can't be applied to the wrapping element the same for all uses. The supports mechanism doesn't allow for the level of customization needed.

An example can be seen for Social Links implementation here: https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/social-links/edit.js#L119

the Social Links uses its own function and specific CSS to apply properly to the correct inner elements.

**Comment by talldan** [comment]
Ok. My comment here still applies - https://github.com/WordPress/gutenberg/pull/30351#issuecomment-814730380

**Comment by jasmussen** [comment]
Alright, to follow up on this one, I rebased to fix the conflict, and I made an additional fix. I realized text in menu items should be right-aligned. Before:

Screenshot 2021-04-13 at 12 23 27

After:

Screenshot 2021-04-13 at 12 32 27

Note that I tested the navigation screen again, and I couldn't see the justification controls there:

Screenshot 2021-04-13 at 12 25 45

So that issue appears fixed. Or was there anything else we needed to do?

**Comment by mkaz** [comment]
@talldan Ok, updated with I think what you were saying. I was confusing props and attributes before.

I went back to previous prop that was in place and switching that to default true, basically does everything we want, plus the additional updates in this PR for style and hiding space-between.

If you can take a look now, I think its right and ready, and sorry for any confusion. 👍

**Review by talldan — APPROVED** [review]
Thanks for making those changes. 🎉 

The only issue I'm seeing when testing is that the spacer block is very narrow for vertical menus, but it doesn't work all that well in `trunk` either (it is wider, but seems to always show `0px` when resizing.)

**Comment by jasmussen** [comment]
> The only issue I'm seeing when testing is that the spacer block is very narrow for vertical menus, but it doesn't work all that well in trunk either (it is wider, but seems to always show 0px when resizing.)

Let me investigate that before I merge.

Thank you for the review!

**Comment by jasmussen** [comment]
In my theme I'm seeing the same before:

Screenshot 2021-04-14 at 09 37 24

Screenshot 2021-04-14 at 09 38 27

and after:

Screenshot 2021-04-14 at 09 39 54
Screenshot 2021-04-14 at 09 40 02

If you have a screenshot or steps to reproduce, I'm happy to take a look.

**Comment by talldan** [comment]
@jasmussen Hmm, not much in the way of repro steps, just added the spacer to a vertical nav block with this branch checked out. Seems the same for Twenty Twenty and Twenty Twenty One (and also the same with theme styles disabled in the editor):
Screenshot 2021-04-14 at 4 52 49 pm

Screenshot 2021-04-14 at 4 53 46 pm

I also notice if you switch between horizontal and vertical nav menus, the spacer seems to retain the width from when it was horizontal. And some `NaN` errors in the console.

Just to check if your styles have rebuilt properly, is your `wp-block-navigation__container` set to `display: flex` when you inspect it? That seems to be the change that made this happen. Switching back to `block` makes the spacer full width again.

**Comment by jasmussen** [comment]
Okay I was able to reproduce! Thank you. Marcus don't merge this until I have a chance to look at it :D

**Comment by jasmussen** [comment]
Okay I restored a property which makes it work for me:

Screenshot 2021-04-14 at 13 28 05

Marcus please test that alignments left right and center as well as the spacer in the vertical nav block work for you also.

**Comment by mkaz** [comment]
This all tested well for me 👍 
Thanks all for the collab and teamwork! 🎉

**Pull request merged**
